yb-voyager/cmd/exportSchema.go

```diff
@@ -260,23 +261,29 @@ func applyShardedTablesRecommendation(shardedTables []string) error {
 	log.Infof("applying colocated vs sharded tables recommendation")
 	var newSQLFileContent strings.Builder
 	sqlInfoArr := parseSqlFileForObjectType(filePath, "TABLE")
-	setOrSelectRegexp := regexp.MustCompile(`(?m)^SET .+?;$|^SELECT .+?;$`)
-	lastStmtSetOrSelect := false
 	for _, sqlInfo := range sqlInfoArr {
-		newSQL := sqlInfo.formattedStmt
-		if setOrSelectRegexp.MatchString(sqlInfo.formattedStmt) {
-			newSQL += "\n"
-			lastStmtSetOrSelect = true
-		} else {
-			if createTableRegex.MatchString(sqlInfo.stmt) &&
-				slices.Contains(shardedTables, sqlInfo.objName) {
-				newSQL = applyShardingRecommendation(sqlInfo, lastStmtSetOrSelect)
-			} else {
-				newSQL = appendSpacing(newSQL, lastStmtSetOrSelect)
+		var modifiedSqlStmt string
+		/*
+			We can rely on pg_query to detect if it is CreateTable and also table name
+			but due to time constraint this module can't be tested thoroughly so relying on the existing as much as possible
+
+			We can pass the whole .sql file as a string also to pg_query.Parse() all the statements at once.
+			But avoiding that also specially for cases where the SQL syntax can be invalid
+		*/
+		// parsing only the Recommended Sharded Tables
+		if sqlInfo.objName != "" && slices.Contains(shardedTables, sqlInfo.objName) {
+			log.Infof("applying sharding recommendation for table=%q", sqlInfo.objName)
+			modifiedSqlStmt, err = applyShardingRecommendation(&sqlInfo)
+			if err != nil {
+				log.Errorf("failed to apply sharding recommendation for table=%q: %v", sqlInfo.objName, err)
+				fmt.Printf("unable to apply sharding recommendation for table=%q, ignoring the recommendation and continue...\n", sqlInfo.objName)
```

Comment:
Shall we also mention "Pls manually add the clause "WITH COLOCATION=false to table <xyz>" ?

Reply:
makes sense. I will add